File: day2/sub.py
f = open('data.txt', 'r')

[hor, vert] = [0, 0]
for x in f:
    [d, i] = x.split(' ')
    if d == "forward":
        hor += int(i)
    elif d == "down":
        vert += int(i)
    elif d == "up":
        vert -= int(i)

print("HOR", hor)
print("VERT", vert)

print(hor * vert)

File: day2/sub.js
const fs = require('fs');

fs.readFile('data.txt', (err, data) => {
    if (err) throw err;

    let lines = data.toString().split('\n').map(x => x.split(' '));
    let [hor, vert] = [0, 0];

    lines.forEach(l => {
        let [dir, a] = [l[0], parseInt(l[1])]
        let funcs = {'forward': (() => hor += a), 
                     'down':    (() => vert += a), 
                     'up':      (() => vert -=a)}
        if(dir in funcs)
            funcs[dir]();
    })

    console.log(hor * vert);
})
